ch.sbb.scion.rcp.microfrontend/src/ch/sbb/scion/rcp/microfrontend/SciRouterOutlet.java

```diff
@@ -120,16 +127,22 @@ private IDisposable installRouter(final String outletName) {
       }
     });
 
-    return () -> subscription.unsubscribe();
+    return subscription::unsubscribe;
   }
 
-  private IDisposable installKeystrokeDispatcher() {
+  private IDisposable installKeystrokeDispatcher(final Control keystrokeTarget) {
+    if (keystrokeTarget == null) {
+      return () -> {
+        // noop
+      };
+    }
     return routerOutletProxy.onKeystroke(event -> {
-      Platform.getLog(SciRouterOutlet.class).info(String.format(//
-          "TODO: Dispatching event to SWT [type=%s, keyCode=%s, character=%s, stateMask=%s]", //
-          Integer.valueOf(event.type), Integer.valueOf(event.keyCode), Character.valueOf(event.character),
-          Integer.valueOf(event.stateMask)));
-      // TODO Dispatching event to SWT (getDisplay().post(event))
+      keystrokeTarget.setFocus();
+      // Prevent infinite cycle:
+      if (browser.isFocusControl()) {
+        throw new IllegalStateException("Browser must not have focus when posting forwarded keystroke event!");

```

Comment:
From a reader pov: What can I do, when I receive this exception? Could the message more focus on what "to do" instead of what "is"?